upt_macports/upt_macports.py

```diff
@@ -176,6 +180,44 @@ def create_package(self, upt_pkg, output=None):
             pkg_cls = pkg_classes[upt_pkg.frontend]
         except KeyError:
             raise upt.UnhandledFrontendError(self.name, upt_pkg.frontend)
-
         packager = pkg_cls()
         packager.create_package(upt_pkg, output)
+
+    def pkg_exist(self, pkg_name, frontend):
+        prefix = {
+            'pypi': 'py-',
+            'rubygems': 'rb-',
+            'cpan': 'p5-',
+        }
+        port_name = prefix.get(self.upt_pkg.frontend, '') + self.upt_pkg.name
+        self.logger.info(f'Checking MaPorts tree for port {port_name}')
+        cmd = f'port info {port_name}'
+        port = os.popen(cmd).read()
```

Comment:
I think we should be using ```subprocess``` here instead of ```os.popen``` (see [here](https://docs.python.org/3/library/subprocess.html#replacing-os-popen-os-popen2-os-popen3)). Currently, if a port isn't found it will print that output to the screen, we should probably just suppress that since we are already logging it.